### PR TITLE
fix(legacy-charts): fix render multiple charts at one page

### PR DESCRIPTION
**Pull request by thabarbados**
### Fix or Enhancement?
fix the error, when multiple charts can't render on one page
from issues #800 #801

- [x] All tests passed

### Environment
- OS: BigSur MacOs
- NPM Version: 8.5.1

### CODE REVIEW

File: legacy/src/Charts.js [hunk 1]
```diff
@@ -22,8 +22,10 @@ import {
   ChartEmits
 } from '../../src/utils'
 
+import { ComponentStore } from './store.ts'
+
 export function generateChart(chartId, chartType, chartController) {
-  let _chart = null
+  let _store = null
```
Comment (dangreen) [inline legacy/src/Charts.js:28]: it looks like "ref", so then maybe just

```
let _chartRef = { current: null }
...
_chartRef.current
```

?

**Review by dangreen — CHANGES_REQUESTED** [review]

File: legacy/src/Charts.js (hunk 1 above)
Comment (thabarbados) [reply legacy/src/Charts.js:28]: @dangreen, I checked, and your solution doesn't fix this bug.

Comment (dangreen) [reply legacy/src/Charts.js:28]: ok, here just `let _chartRef = null`

`_store = new ComponentStore()` -> `_chartRef = { current: null }`
`_store.setComponentChart(...)` -> `_chartRef.current = ...`
etc

Comment (thabarbados) [reply legacy/src/Charts.js]: It works.  I used it.

File: legacy/src/Charts.js [hunk 2]
```diff
@@ -80,8 +82,12 @@ export function generateChart(chartId, chartType, chartController) {
     },
     methods: {
       renderChart(data, options) {
-        if (_chart !== null) {
-          chartDestroy(_chart)
+        if (
+          _chartRef !== null &&
+          'current' in _chartRef &&
+          _chartRef.current !== null
```
Comment (dangreen) [inline legacy/src/Charts.js:88]:
```suggestion
          _chartRef?.current
```

?

**Review by dangreen — CHANGES_REQUESTED** [review]

File: legacy/src/Charts.js [hunk 3]
```diff
@@ -92,8 +98,12 @@ export function generateChart(chartId, chartType, chartController) {
 
           const canvasEl2DContext = this.$refs.canvas.getContext('2d')
 
-          if (canvasEl2DContext !== null) {
-            _chart = new ChartJS(canvasEl2DContext, {
+          if (
+            canvasEl2DContext !== null &&
+            _chartRef !== null &&
+            'current' in _chartRef
```
[the remark this answers is not in the document]
Comment (thabarbados) [reply legacy/src/Charts.js]: removed

**Review by dangreen — APPROVED** [review]

**Pull request merged**